### DISCUSSION TITLE
Remember session (or device) volume

### DISCUSSION THREAD

**Discussion by spmp** [mikebrady/shairport-sync#1896] [Q&A]
Shairport-sync is AMAZING! Thank you. So easy and intuitive. Just works like it says on the box, maybe more 8)

Whenever I connect to the `shairport-sync` AirPlay device the volume resets to about half, even if this is within the session time of 120s (i.e switch audio to iPhone, then back to Shairport).
Is there a way to have `shairport-sync` remember the session or device volume and keep this persistent between or within a session?

Chur

**Comment by mikebrady** [comment]
Thanks for the kind words!

Could you just run:
```
$ shairport-sync --displayConfig
```
and post the result please? Just so that we know the configuration of the system.

Also, what are you playing from, e.g. Apple Music on iOS, etc.

**Comment by spmp** [comment]
My version of `shairport-sync` doesn't have that command. It's version: `3.3.8-libdaemon-OpenSSL-Avahi-ALSA-jack-pa-dummy-stdout-pipe-soxr-convolution-metadata-mqtt-dbus-mpris-sysconfdir:/etc`
from Ubuntu 24.04.

I've made the following changes to stock configuration:
```
general = { 
  name = "The Lighthouse 🤗";
  output_backend = "pa";
...
```
And added
```
PULSE_SERVER=192.168.1.X
```
To systemd env-vars (I know, can put it in three config...)

I am playing from an iPhone 12 mini, iOS version 17.5.1. I'm certain when
playing to genuine Apple TV the volume was persistent 😉

Is there a PPA which would bump me to a newer version? Going to stick with
24.04 as it'll LTS soon.

Cheers

**Comment by mikebrady** [comment]
Thanks. You've answered enough of the question -- it has to be "classic" AirPlay, and you're playing from an up-to-date iOS. let me check the situation -- AFAIR, the iPhone resets the volume to that low level, but I need to check.

**Reply by spmp** [reply]
With the latest version built from source it works as expected out of the box. Same config.
Volume preserved for device, not just within a 120s session.

**Comment by spmp** [comment]
I know this is off topic, but WOW, the latest version of `shairport-sync` does everything so well - I can make calls and get notifications and the music still plays... Just like a bought one 8)

I am taking the recommendation to use `podman` to deploy containers as `systemd` services, happy to contribute documentation, but I am far from an expert.

**Reply by mikebrady** [reply]
Great -- please let us know how you get on!
